include/library/spdm_device_secret_lib.h

```diff
@@ -23,19 +23,22 @@
                                        It must align with measurement_specification (SPDM_MEASUREMENT_BLOCK_HEADER_SPECIFICATION_*)
   @param  measurement_hash_algo          Indicates the measurement hash algorithm.
                                        It must align with measurement_hash_algo (SPDM_ALGORITHMS_MEASUREMENT_HASH_ALGO_*)
-  @param  device_measurement_count       The count of the device measurement block.
-  @param  device_measurement            A pointer to a destination buffer to store the concatenation of all device measurement blocks.
-  @param  device_measurement_size        On input, indicates the size in bytes of the destination buffer.
+  @param  measurement_index      The index of the measurement to collect.
+  @param  measurements_count       The count of the device measurement block.
+  @param  measurements            A pointer to a destination buffer to store the concatenation of all device measurement blocks.
+  @param  measurements_size        On input, indicates the size in bytes of the destination buffer.
                                        On output, indicates the size in bytes of all device measurement blocks in the buffer.
 
-  @retval TRUE  the device measurement collection success and measurement is returned.
-  @retval FALSE the device measurement collection fail.
+  @retval RETURN_SUCCESS             Successfully returned measurement_count and optionally measurements, measurements_size.
+  @retval RETURN_BUFFER_TOO_SMALL    "measurements" buffer too small for measurements.
+  @retval RETURN_INVALID_PARAMETER   Invalid parameter passed to function.
+  @retval RETURN_***                 Any other RETURN_ error from base.h
 **/

```

Comment:
I updated this function pointer typedef to match the spdm_measurement_collection function, but it seems that this function pointer typedef, and a few others like it, are never used. Perhaps we should remove it in another pull request, since it seemed we were interested in removing function pointers for security.